js/src/main/scala/doodle/js/Svg.scala

```diff
@@ -34,13 +54,57 @@ object Svg {
     }
 
     dc.fill.fold(builder ++= "fill: none; ") {
-      case Fill(color) =>
+      case Fill.Color(color) =>
         builder ++= s"fill: ${toHSLA(color)}; "
+      case Fill.Gradient(gradient) =>
+        builder ++= s"fill: url(#${toGradientId(gradient)}); "
     }
 
     builder.toString
   }
 
+  def toSvgSpreadMethod(cycleMethod: Gradient.CycleMethod): String = cycleMethod match {
+    case _: Gradient.CycleMethod.NoCycle => "pad"
```

Comment:
Following suggested change to `case object` these patterns matches should match directly on the objects, which I find clearer.